internal/transport/http2_client.go

```diff
@@ -241,7 +241,7 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 			if cd.RequireTransportSecurity() {
 				if ci, ok := authInfo.(internalInfo); ok {
 					secLevel := ci.GetCommonAuthInfo().SecurityLevel
-					if secLevel != credentials.Invalid && secLevel < credentials.PrivacyAndIntegrity {
+					if secLevel != credentials.InvalidSecurityLevel && secLevel < credentials.PrivacyAndIntegrity {
```

Comment:
I was thinking we could use `CheckSecurityeLevel` here, but I see this is different from `CheckSecurityLevel` because `Invalid` is not okay here.  LGTM, but please move `internalInfo` so it's defined immediately above to limit its scope (or use it inline).